smartsim/_core/generation/operations.py

```diff
@@ -0,0 +1,298 @@
+import base64
+import os
+import pathlib
+import pickle
+import sys
+import typing as t
+from dataclasses import dataclass, field
+
+from ..commands import Command
+
+entry_point_path = "smartsim._core.entrypoints.file_operations"
+"""Path to file operations module"""
+
+copy_cmd = "copy"
+"""Copy file operations command"""
+symlink_cmd = "symlink"
+"""Symlink file operations command"""
+configure_cmd = "configure"
+"""Configure file operations command"""
+
+
+def create_final_dest(
+    job_root_path: pathlib.Path, dest: t.Union[pathlib.Path, None]
+) -> str:
+    """Combine the job root path and destination path. Return as a string for
+    entry point consumption.
+
+    :param job_root_path: Job root path
+    :param dest: Destination path
+    :return: Combined path
+    :raises ValueError: An error occurred during path combination
+    """
+    try:
+        combined_path = job_root_path
+        if dest:
+            combined_path = job_root_path / dest
+        return str(combined_path)
+    except Exception as e:
+        raise ValueError(f"Error combining paths: {e}")
```

Comment:
Just to make the stack trace a bit nicer
```suggestion
    except Exception as e:
        raise ValueError(f"Error combining paths: {e}") from e
```